cdap-api/src/main/java/co/cask/cdap/api/mapreduce/MapReduceSpecification.java

```diff
@@ -59,4 +62,19 @@
    */
   @Nullable
   Resources getReducerResources();
+
+  /**
+   * @return Map of streams and {@link StreamSpecification} created in this program.
+   */
+  Map<String, StreamSpecification> getStreams();
```

Comment:
these 3 methods also implemented in a bunch of places, is there a reason they aren't shared?